pkg/kubelet/kubelet.go

```diff
@@ -3286,6 +3286,16 @@ func (kl *Kubelet) generateAPIPodStatus(pod *api.Pod, podStatus *kubecontainer.P
 	s.Phase = GetPhase(spec, s.ContainerStatuses)
 	kl.probeManager.UpdatePodStatus(pod.UID, s)
 	s.Conditions = append(s.Conditions, status.GeneratePodReadyCondition(spec, s.ContainerStatuses, s.Phase))
+	// s (the PodStatus we are creating) will not have a PodScheduled condition yet, because converStatusToAPIStatus()
+	// does not create one. If the existing PodStatus has a PodScheduled condition, then copy it into s and make sure
+	// it is set to true. If the existing PodStatus does not have a PodScheduled condition, then create one that is set to true.
```

Comment:
Questions (without reading the entire PR): Should `PodScheduled` condition be set even if a pod doesn't go through the scheduler? E.g., (1) user writes the "Node" field in the pod spec directly, and (2) pods from non-apiserver sources. The condition doesn't mean much at all in these cases. Why can't kubelet simply copy the existing condition if there is one?